### ISSUE TITLE
[BUG] Nested pydantic model in response model doesn't filter additional data

### ISSUE THREAD

**Issue by sagrawal-idrc** [tiangolo/fastapi#847]
### Describe the bug

In [Response Model documentation](https://fastapi.tiangolo.com/tutorial/response-model/), there is an example for `FastAPI will take care of filtering out all the data that is not declared in the output model`. However, this doesn't happen for a nested model.

### To Reproduce

1. Create a file with:

```Python
from fastapi import FastAPI, Depends
from pydantic import BaseModel

app = FastAPI()


class ModelB(BaseModel):
    username: str


class ModelC(ModelB):
    password: str


class ModelA(BaseModel):
    name: str
    description: str = None
    model_b: ModelB


async def get_model_c() -> ModelC:
    return ModelC(username="test-user", password="test-password")


@app.get("/model", response_model=ModelA)
async def get_model_a(model_c=Depends(get_model_c)):
    return {"name": "model-a-name", "description": "model-a-desc", "model_b": model_c}
```

3. Open the browser and call the endpoint `/model`.
4. It returns a JSON with `{"name":"model-a-name","description":"model-a-desc","model_b":{"username":"test-user","password":"test-password"}}`.
5. But I expected it to return `{"name":"model-a-name","description":"model-a-desc","model_b":{"username":"test-user"}}`.

### Expected behavior

Filtering should be done for nested model attributes too.

### Environment

- OS: Ubuntu 18.04.3 LTS
- FastAPI 0.45.0
- Python 3.8.1

**Comment by chbndrhnns** [comment]
I see this related to https://github.com/tiangolo/fastapi/issues/800

**Comment by dmontagu** [comment]
Yep, this really needs to be fixed. I haven't had much free time lately but I'll try to get around to fixing this soon.

As a short term workaround, it should work properly if you call `return jsonable_encoder(<current response>)` where jsonable_encoder is `fastapi.encoders.jsonable_encoder`. But obviously it would be better if this wasn't required.

**Comment by tiangolo** [comment]
Thanks for the easy to replicate and test report @sagrawal-idrc ! :bug: 

And thanks @dmontagu for the help here :muscle: 

And yeah, @chbndrhnns , it was the same issue.

It should be fixed by #889, just released in the latest version `0.47.1`.
